### PR TITLE
Add more information to failed results

### PR DESCRIPTION
**Pull request by ngie-eign**
This change adds more information to "premature failures" or failures where no output was provided on the command-line. This is being done to help make the reasons for why a failure occurred clearer to the end-user.

This also fixes the fact that several ATF-related tests were failing with "empty output" (lacking exit code data), or not failing with the appropriate (expected) "timeout" messages.

While here, address the fact that newer versions of sqlite3 fail when  executing statements with `sqlite::api_error` instead of `store::error`.

This addresses part of #237 and makes the kyua tests pass cleanly again on FreeBSD.

### CODE REVIEW

File: engine/atf_result.cpp [hunk 1]
```diff
@@ -625,13 +625,21 @@ engine::calculate_atf_result(const optional< process::status >& body_status,
...
         result = atf_result::load(results_file);
     } catch (const engine::format_error& error) {
-        result = atf_result(atf_result::broken, error.what());
+        if (body_status) {
+            result = atf_result(atf_result::broken,
+                F("Error: %s. Test case %s") %
+                error.what() %
+                format_status(body_status.get()));
+        } else {
+            // The test case timed out.  apply() handles this case later.
```
Comment (emaste) [inline engine/atf_result.cpp:634]: This seems a bit strange style-wise, but is already like this elsewhere so being consistent makes sense for now.

File: store/write_transaction_test.cpp [hunk 2]
```diff
@@ -137,8 +137,19 @@ ATF_TEST_CASE_BODY(commit__fail)
         backend.database().exec(
             "CREATE TABLE foo ("
             "a REFERENCES env_vars(var_name) DEFERRABLE INITIALLY DEFERRED)");
-        backend.database().exec("INSERT INTO foo VALUES (\"WHAT\")");
+        // For whatever reason, multiple Linux distros seem to execute the
```
Comment (emaste) [inline store/write_transaction_test.cpp:140]: This looks like something that someone (tm) should do a deep dive into, sometime.

**Review by emaste — APPROVED** [review]

File: engine/atf_result.cpp (hunk 1 above)
Comment (ngie-eign) [reply engine/atf_result.cpp:634]: Yes. The atf/kyua/lutok coding style doesn't match FreeBSD style(9).

File: store/write_transaction_test.cpp (hunk 2 above)
Comment (ngie-eign) [reply store/write_transaction_test.cpp:140]: Thank you for the reminder! I'm going to file the issue after I merge this change :).
Also, FWIW... MacOS fails in the same way that FreeBSD does. I'll have to check and see if NetBSD is affected by this at all...

Comment (ngie-eign) [reply store/write_transaction_test.cpp:140]: Opened #240.
